### PR TITLE
Add `openApp` method

### PR DESCRIPTION
**Pull request by Kudo**
`open` is good to facilitate cross platform opening. sometimes i just want to open an application without specifying url or file target. the concept is something like `open(null, { app: { name: open.apps.chrome }});`. this pr just introduces an `openApp` api. please help to review if the concept makes sense.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -69,11 +69,7 @@ const pTryEach = async (array, mapper) => {
 	throw latestError;
 };
 
-const open = async (target, options) => {
-	if (typeof target !== 'string') {
-		throw new TypeError('Expected a `target`');
-	}
-
+const _openImpl = async options => {
```
Comment (sindresorhus) [inline index.js:72]:
```suggestion
const baseOpen = async options => {
```

File: index.d.ts [hunk 2]
```diff
@@ -115,6 +115,32 @@ declare const open: {
...
+	await openApp(apps.firefox);
+
+	// Open Chrome incognito mode
+	await openApp(apps.chrome, ['--incognito']);
+
+	// Open Xcode
+	await openApp('xcode');
+	```
+	*/
+	openApp: (name: string, appArguments?: string[], options?: open.Options) => Promise<ChildProcess>;
```
Comment (sindresorhus) [inline index.d.ts:143]: It should not allow the `app` option. You can use the `Omit` helper for this.

File: index.d.ts [hunk 3]
```diff
@@ -115,6 +115,32 @@ declare const open: {
 	```
 	*/
 	apps: Record<open.AppName, string | readonly string[]>;
+
+	/**
+	Open an application. Cross-platform.
```
Comment (sindresorhus) [inline index.d.ts:120]:
```suggestion
	Open an app. Cross-platform.
```

Use the word `app` in other places too.

File: index.d.ts [hunk 4]
```diff
@@ -115,6 +115,32 @@ declare const open: {
...
+
+	Uses the command `open` on macOS, `start` on Windows and `xdg-open` on other platforms.
+
+	There is a caveat for [double-quotes on Windows](https://github.com/sindresorhus/open#double-quotes-on-windows) where all double-quotes are stripped from the `target`.
```
Comment (sindresorhus) [inline index.d.ts:124]: Not relevant to this method.

File: readme.md [hunk 5]
```diff
@@ -40,6 +40,12 @@ await open('https://sindresorhus.com', {app: {name: 'firefox'}});
 
 // Specify app arguments.
 await open('https://sindresorhus.com', {app: {name: 'google chrome', arguments: ['--incognito']}});
+
+// Open app
```
Comment (sindresorhus) [inline readme.md:44]:
```suggestion
// Open an app
```

File: readme.md [hunk 6]
```diff
@@ -40,6 +40,12 @@ await open('https://sindresorhus.com', {app: {name: 'firefox'}});
...
+await open.openApp('xcode');
+
+// Open app with arguments
```
Comment (sindresorhus) [inline readme.md:47]:
```suggestion
// Open an app with arguments
```

File: readme.md [hunk 7]
```diff
@@ -130,6 +136,24 @@ await open('https://google.com', {
 - [`firefox`](https://www.mozilla.org/firefox) - Web browser
 - [`edge`](https://www.microsoft.com/edge) - Web browser
 
+### open.openApp(name, appArguments?, options?)
+
```
Comment (sindresorhus) [inline readme.md:140]: Should describe what it does.

File: readme.md [hunk 8]
```diff
@@ -130,6 +136,24 @@ await open('https://google.com', {
 - [`firefox`](https://www.mozilla.org/firefox) - Web browser
 - [`edge`](https://www.microsoft.com/edge) - Web browser
 
+### open.openApp(name, appArguments?, options?)
```
Comment (sindresorhus) [inline readme.md:139]: I would prefer the `appArguments` to be in options.

File: readme.md [hunk 9]
```diff
@@ -130,6 +136,24 @@ await open('https://google.com', {
...
+
+Returns a promise for the [spawned child process](https://nodejs.org/api/child_process.html#child_process_class_childprocess). You would normally not need to use this for anything, but it can be useful if you'd like to attach custom event listeners or perform other operations directly on the spawned process.
+
+#### name
+
```
Comment (sindresorhus) [inline readme.md:144]: These should include `Type` like the other documented parameters and options.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by Kudo** [comment]
thanks @sindresorhus. these review comments are neat and i should update all. please take a look when you get a chance.

File: index.d.ts [hunk 10]
```diff
@@ -54,6 +54,15 @@ declare namespace open {
 		readonly allowNonzeroExitCode?: boolean;
 	}
 
+	interface OpenAppOptions extends Omit<Options, 'app'> {
+		/**
+		Optional arguments to open.
```
Comment (sindresorhus) [inline index.d.ts:59]: This description is not really correct.
```suggestion
		Arguments passed to the app.
```

File: index.d.ts [hunk 11]
```diff
@@ -54,6 +54,15 @@ declare namespace open {
...
+
+		The app `arguments` are app dependent. Check the app's documentation for what arguments it accepts.
```
Comment (sindresorhus) [inline index.d.ts:61]:
```suggestion
		These arguments are app dependent. Check the app's documentation for what arguments it accepts.
```

File: index.d.ts [hunk 12]
```diff
@@ -54,6 +54,15 @@ declare namespace open {
...
+		 */
```
Comment (sindresorhus) [inline index.d.ts:62]:
```suggestion
		*/
```

File: index.d.ts [hunk 13]
```diff
@@ -54,6 +54,15 @@ declare namespace open {
...
+		readonly arguments?: string[];
```
Comment (sindresorhus) [inline index.d.ts:63]:
```suggestion
		readonly arguments?: readonly string[];
```

File: readme.md [hunk 14]
```diff
@@ -130,6 +136,33 @@ await open('https://google.com', {
...
+The app name is platform dependent. Don't hard code it in reusable modules. For example, Chrome is `google chrome` on macOS, `google-chrome` on Linux and `chrome` on Windows. If possible, use [`open.apps`](#openapps) which auto-detects the correct binary to use.
+
+You may also pass in the app's full path. For example on WSL, this can be `/mnt/c/Program Files (x86)/Google/Chrome/Application/chrome.exe` for the Windows installation of Chrome.
+
+#### options
+
+Type: `object`
+
+The options are like the options in [`open`](#options) except to remove `options.app` and add the following options.
```
Comment (sindresorhus) [inline readme.md:157]:
```suggestion
Same options as [`open`](#options) except `app` and with the following additions:
```

File: readme.md [hunk 15]
```diff
@@ -130,6 +136,33 @@ await open('https://google.com', {
...
+
+##### arguments
+
+Type: `Array<string>`
```
Comment (sindresorhus) [inline readme.md:161]:
```suggestion
Type: `string[]`\
```

File: readme.md [hunk 16]
```diff
@@ -130,6 +136,33 @@ await open('https://google.com', {
...
+Default: `[]`
+
```
Comment (sindresorhus) [inline readme.md:163]: Missing the description from index.d.ts

**Comment by sindresorhus** [comment]
Can you manually verify that these changes do not break normal usage of `open`?

**Comment by Kudo** [comment]
i did go through the test cases one by one in macos. there are just two cases with pipe not opening anything. that's same with main branch and `open` does not encode `target` in macos. 
https://github.com/sindresorhus/open/blob/e21e623d459210f899dba34bceea2dfdbb6f3827/test.js#L57-L63

**Pull request merged**